include/ruby/internal/fl_type.h

```diff
@@ -165,9 +168,29 @@ ruby_fl_type {
     RUBY_FL_PROMOTED1    = (1<<6),
     RUBY_FL_PROMOTED     = RUBY_FL_PROMOTED0 | RUBY_FL_PROMOTED1,
     RUBY_FL_FINALIZE     = (1<<7),
-    RUBY_FL_TAINT        = (1<<8),
+    RUBY_FL_TAINT
+
+#if RBIMPL_HAS_EXTENSION(enumerator_attributes)
+    RBIMPL_ATTR_DEPRECATED(("taintedness turned out to be a wrong idea."))
+#elif RBIMPL_COMPILER_SINCE(GCC, 6, 0, 0)
+    RBIMPL_ATTR_DEPRECATED(("taintedness turned out to be a wrong idea."))
```

Comment:
Why repeating the same message?

Reply:
Because GCC 6 doesn't understand `__has_extension()`, we have to do things once for Clang and once for GCC.

Reply:
That means `RBIMPL_HAS_EXTENSION(enumerator_attributes) || RBIMPL_COMPILER_SINCE(GCC, 6, 0, 0)` doesn't work?

Reply:
Yes, that should work.  That line would be a bit too long to me though.  It's just a matter of preference.  I like 80-ish columns.

Reply:
https://github.com/ruby/ruby/pull/4100/commits/bb668e13e10c38dadcf8190078ef22be2c5d99df

Reply:
I agree that it is too long.
OTOH, repeating same blocks three times also feels redundant to me.
I thought a macro for deprecated enums, such as `RBIMPL_DEPRECATED_ENUM(RUBY_FL_TAINT, ("taintedness turned out to be a wrong idea."))`, but MSC doesn't support `_Pragma`?

Reply:
Seems nice.
Yet another question came out.
`RBIMPL_ATTR_DEPRECATED` seems always defined here, doesn't it work unconditionally?

Reply:
No, some old GCC don't understand `enum { foo __attribute__((deprecated)) };`.

Reply:
I see, thank you.